### PR TITLE
Added a Set of ESLint Rules  and remove warnings

### LINKED ISSUE #15: Add a Set of ESLint Rules and Clear the Errors/Warning
> Before attempting this issue, please share the set of the rules you're planning to put

We want to have better and more consistent linting rules in the tool that will not balance productivity with efficiency. This task is about having ESLint implemented in the repository with a set of rules that fit the application and cleaning the current warnings.

Acceptance criteria:
- Implement set of rules and clear the current warning/errors in the code
- Add a command to run the linting
- Share a video with sanity check of all the features in the CLI (so we know they haven't broken)

### PR DESCRIPTION
**Pull request by Mubashirshariq**
closes #15 
/claim #15

### CODE REVIEW

**Comment by Mubashirshariq** [comment]
@gemanor  Please let me know if you want any other rules then i will add those also

**Comment by Mubashirshariq** [comment]
https://github.com/user-attachments/assets/3bc2f6c2-0014-4aca-881b-ef3be5fbe6f6

**Comment by Mubashirshariq** [comment]
@quest-bot loot #15

**Comment by Mubashirshariq** [comment]
@gemanor  i have made the requested changes ,plz review

File: package.json [hunk 1]
```diff
@@ -28,21 +30,32 @@
 		"ink-spinner": "^5.0.0",
 		"keytar": "^7.9.0",
 		"open": "^10.1.0",
+		"parser": "^0.1.4",
```
Comment (gemanor) [inline package.json:33]: Why do we need it as a dependency and not a dev dependency?

File: tsconfig.json [hunk 2]
```diff
@@ -6,5 +6,6 @@
 		"module": "Node16",
 		"outDir": "dist"
 	},
-	"include": ["source"]
+	"include": ["source/**/*"],
```
Comment (gemanor) [inline tsconfig.json:9]: Why changing the `include` part?

File: eslint.config.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+import prettierConfig from 'eslint-config-prettier';
+
+
+const compat = new FlatCompat({
+  baseDirectory: import.meta.url,
+});
+
+const eslintConfig = [js.configs.recommended,
+	...compat.extends('plugin:react/recommended'),
+  prettierConfig
```
Comment (gemanor) [inline eslint.config.js:18]: Need to fix spacing/formating in this file

File: package.json [hunk 4]
```diff
@@ -12,6 +12,8 @@
 	"scripts": {
 		"build": "tsc",
 		"dev": " NODE_NO_WARNINGS=1 tsc --watch",
+		"lint": "eslint \"source/**/*.{js,ts,tsx}\" --fix",
```
Comment (gemanor) [inline package.json:15]: It should be switched. `lint` will do check, and `lint:fix` will do check+fix.

File: source/commands/apiKey.tsx [hunk 5]
```diff
@@ -22,21 +22,21 @@ type Props = {
...
+export default function ApiKey({ args, options }: Props) {
 	const action: string = args[0];
 	const key: string = args[1];
 	const isValid = key.length >= 97 && key.startsWith('permit_key_');
 
 	const [readKey, setReadKey] = React.useState('');
 
-	if (action === 'read') {
-		React.useEffect(() => {
+	React.useEffect(() => {
```
Comment (gemanor) [inline source/commands/apiKey.tsx:32]: Please import from React instead of React.useEffect.

File: source/commands/login.tsx [hunk 6]
```diff
@@ -31,22 +31,36 @@ export const options = object({
 		),
 });
```
Comment (gemanor) [inline source/commands/login.tsx:32]: Please attach sanity test recording for the login flow.

**Review by gemanor — COMMENTED** [review]
Looks good. Please address the attached comments.

**Comment by Mubashirshariq** [comment]
@gemanor , Suggested changes has been implemented. Plz review it.


https://github.com/user-attachments/assets/6e66ec04-0811-46f9-b8ac-f216ce9f335b

**Comment by Mubashirshariq** [comment]
@gemanor could you please take a look, bounty for this issue is not released yet.Thanks

**Pull request merged**